DigitalLearningSolutions.Data/Services/DelegateUploadFileService.cs

```diff
@@ -0,0 +1,222 @@
+                    break;
+                case "-2":
+                    errors.Add((delegateRow.RowNumber, BulkUploadResult.ErrorReason.ParameterError));
+                    break;
+                case "-3":
+                    errors.Add((delegateRow.RowNumber, BulkUploadResult.ErrorReason.AliasIdInUse));
+                    break;
+                case "-4":
+                    errors.Add((delegateRow.RowNumber, BulkUploadResult.ErrorReason.EmailAddressInUse));
+                    break;
+                default:
+                    registered += 1;
+                    break;
+            }
+        }
+
+        private void UpdateDelegate(
+            DelegateTableRow delegateRow,
+            bool approved,
+            int centreId,
+            ref int updated,
+            ref int skipped,
+            ref int registered,
+            List<(int, BulkUploadResult.ErrorReason)> errors
+        )
+        {
+            var record = new DelegateRecord(delegateRow, centreId, approved);
+            var status = userDataService.UpdateDelegateRecord(record);
+            switch (status)
+            {
+                case 0:
+                    updated += 1;
+                    break;
+                case 1:
+                    skipped += 1;
+                    break;
+                case 2:
+                    registered += 1;
+                    break;
+                case -1:
+                case -4:
+                    errors.Add((delegateRow.RowNumber, BulkUploadResult.ErrorReason.UnexpectedErrorForUpdate));
+                    break;
+                case -2:
+                    errors.Add((delegateRow.RowNumber, BulkUploadResult.ErrorReason.ParameterError));
+                    break;
+                case -3:
+                    errors.Add((delegateRow.RowNumber, BulkUploadResult.ErrorReason.AliasIdInUse));
+                    break;
+                case -5:
+                    errors.Add((delegateRow.RowNumber, BulkUploadResult.ErrorReason.EmailAddressInUse));
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException(
+                        nameof(status),
+                        status,
+                        "Unknown return value when updating delegate record."
+                    );
+            }
+        }
+
+        private static bool ValidateHeaders(IXLTable table)
+        {
+            var expectedHeaders = new List<string>
+            {
+                "LastName",
+                "FirstName",
+                "DelegateID",
+                "AliasID",
+                "JobGroupID",
+                "Answer1",
+                "Answer2",
+                "Answer3",
+                "Answer4",
+                "Answer5",
+                "Answer6",
+                "Active",
+                "EmailAddress"
+            }.OrderBy(x => x);

```

Comment:
Currently I don't think this is necessary since it is only used to validate the headers from the file here, but if we need them elsewhere then yeah they should.